### DISCUSSION TITLE
How to autostart the windows service using node-windows

### DISCUSSION THREAD

**Discussion by Kriess2507** [coreybutler/node-windows#347] [Q&A]
How can we automatically start the windows service, even after the system reboot using the node-windows library. Also how can we configure an id along with the name for our service

**Comment by Kriess2507** [comment]
@coreybutler could you provide some insights here?

**Comment by coreybutler** [comment]
There is no way to programmatically define autostart behavior with this library, but you can do this directly in Windows using `Set-Service [service name] -startuptype automatic`. So, there's nothing stopping you from running this as a child process in your code, assuming your user account has permissions to do this.

As for service names, you just use the `name` property as shown at https://github.com/coreybutler/node-windows#windows-services. There aren't ID's, unless you're referring to PID's, which Windows generates automatically.

**Reply by Kriess2507** [reply]
Thanks @coreybutler for the immediate response. If I  am not wrong there is a property in winsw called startmode(or something similar) which can be set to automatic. I was wondering if this can be extended to the node-windows library.
As for the ID and the name thing I was thinking on the lines of @miquelfire, as of today bot the name and the display name refers to the same name property. If at all there is any way to specify or rather distinguish between the name and the display name??

**Comment by miquelfire** [comment]
The ID and name they might be referring to is the service name and the display name

If I'm correct on this assumption, the ID would be the service, so for example, the service called "Windows Time" (display name) has the service name of W32Time

It only really matters if you are using something like the net start|stop command or the like.
